Add chebi 20 dataset + Hugging Face datasets example

jackapbutler opened the pull request:
# Overview
* closes #63 

Hey, I've added a `transform.py` script and related `meta.yaml` for the Chebi-20 dataset. It contains a dataset of SMILES with their corresponding natural language description.  The dataset was first used in [this paper](https://aclanthology.org/2021.emnlp-main.47/) but also used [here](https://arxiv.org/pdf/2204.11817.pdf) by the same authors. It also features in Kevin's [awesome-chemistry-datasets](https://github.com/kjappelbaum/awesome-chemistry-datasets) repository.

# Template
I think this might be useful for augmenting some tabular templates with additional context through the natural language descriptions or creating a template itself out of this dataset such as 
```
The molecule with smiles identifier of <SMILES> can be described as follows _______.
```

# Hugging Face - datasets
I used the Hugging Face datasets repository to store the raw csv's at [this url](https://huggingface.co/datasets/OpenBioML/chebi_20) allowing my `transform.py` script to just call this link when downloading the data.

It was quite handy and I think there are a few nice features in the `datasets` package which would make our lives a bit easier on the dataset side of things such as 
* automated multiprocessing for data cleaning
* version controlling the dataset as well as our code
* easy interface into tokenisation and dataloading aspects for model training
* single object (Hugging Face dataset) allows for reuse of utility functions that clean different properties of the dataset

We could consider using this as a pattern for some of our larger datasets? or all if it would help standardise things?

# Questions
I've a a question around the "license" in the `meta.yml` configuration. I wasn't able to find any explicit license inside the research papers or github repositories mentioned above although I think it's possible I'm not looking in the right place. 

The authors has extracted data from PubMed and [Chebi](https://www.ebi.ac.uk/chebi/) to create the dataset, is it possible to use the licenses that the raw data came under?

kjappelbaum commented:
Thanks for this addition! 

I'm not sure I fully get 

> think this might be useful for augmenting some tabular templates with additional context through the natural language descriptions or creating a template itself out of this dataset such as "The molecule with smiles identifier of <SMILES> can be described as follows _______."

The current `meta.yaml` syntax should allow specifying such templates, and yours is a great example. If it is not clear how this is supposed to work, I'm happy to commit an example.

kjappelbaum commented:
>We could consider using this as a pattern for some of our larger datasets? or all if it would help standardise things?

Yes, I'm very open to that! Good idea, we would only need to go through the datasets we already have in the repo.

kjappelbaum commented:
I asked on the GitHub about the license https://github.com/cnedwards/text2mol/issues/2

kjappelbaum commented on data/chebi_20/meta.yaml:
```diff
@@ -0,0 +1,33 @@
+  description: Text2Mol data repository.
+- url: https://huggingface.co/datasets/OpenBioML/chebi_20
+  description: Hugging Face dataset for OpenBioML.
+num_points: null
```
why null?

jackapbutler replied:
Yes, I added the wrong key, it's changed to only having `num_points` now

kjappelbaum commented on data/chebi_20/meta.yaml:
```diff
@@ -0,0 +1,33 @@
+              year={2021},
+              url = {https://aclanthology.org/2021.emnlp-main.47/}
+              }
+num_samples: 3301
```
now https://github.com/OpenBioML/chemnlp/pull/65/files#r1126116143 makes perhaps a bit more sense. But we probably should keep only one. 

Was this incorrect in one of the examples?

jackapbutler commented:
> Thanks for this addition!
> 
> I'm not sure I fully get
> 
> > think this might be useful for augmenting some tabular templates with additional context through the natural language descriptions or creating a template itself out of this dataset such as "The molecule with smiles identifier of  can be described as follows _______."
> 
> The current `meta.yaml` syntax should allow specifying such templates, and yours is a great example. If it is not clear how this is supposed to work, I'm happy to commit an example.

I was just checking if I understood some possible templates for the dataset, great to hear 👍🏽

jackapbutler commented:
Hey @kjappelbaum, I've updated the PR as per your comments. As my dataset has 3 original splits used by the previous authors I've added a `.yaml` for each of those.

I'm happy to call sometime this week about Hugging Face if that suits you?
